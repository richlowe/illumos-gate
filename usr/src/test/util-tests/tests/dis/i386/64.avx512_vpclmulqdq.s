/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2020 Robert Mustacchi
 */

/*
 * Test AVX512 vpclmulqdq related instructions
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	vpclmulqdq	$0x23, %ymm10, %ymm11, %ymm12
	vpclmulqdq	$0x42, 0x12345(%eax), %ymm13, %ymm14
	vpclmulqdq	$-0x17, 0x678(%ebx, %ecx, 4), %ymm15, %ymm16

	vpclmulqdq	$0x23, %zmm12, %zmm11, %zmm10
	vpclmulqdq	$0x42, 0x12345(%eax), %zmm8, %zmm8
	vpclmulqdq	$-0x17, 0x678(%ebx, %ecx, 4), %zmm7, %zmm6
.size libdis_test, [.-libdis_test]
