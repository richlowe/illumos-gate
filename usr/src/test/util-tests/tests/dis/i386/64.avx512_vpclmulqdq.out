    libdis_test:      c4 43 25 44 e2 23  vpclmulqdq $0x23,%ymm10,%ymm11,%ymm12
    libdis_test+0x6:  67 c4 63 15 44 b0  addr32 vpclmulqdq $0x42,0x12345(%rax),%ymm13,%ymm14
                      45 23 01 00 42 
    libdis_test+0x11: 67 62 e3 05 28 44  addr32 vpclmulqdq $-0x17,0x678(%rbx,%rcx,4),%ymm15,%ymm16	<0xe9>
                      84 8b 78 06 00 00 
                      e9 
    libdis_test+0x1e: 62 53 25 48 44 d4  vpclmulqdq $0x23,%zmm12,%zmm11,%zmm10
                      23 
    libdis_test+0x25: 67 62 73 3d 48 44  addr32 vpclmulqdq $0x42,0x12345(%rax),%zmm8,%zmm8
                      80 45 23 01 00 42 
    libdis_test+0x31: 67 62 f3 45 48 44  addr32 vpclmulqdq $-0x17,0x678(%rbx,%rcx,4),%zmm7,%zmm6	<0xe9>
                      b4 8b 78 06 00 00 
                      e9 
